test/junit/scala/collection/immutable/LazyListLazinessTest.scala

```diff
@@ -693,6 +693,20 @@ class LazyListLazinessTest {
     assertLazyAllSkipping(op, 4)
   }
 
+  private def genericCons_unapply_properlyLazy(unapply: LazyList[Int] => Option[(Int, LazyList[Int])]): Unit = {
```

Comment:
I noticed there were no tests for these while I was in the area, so I added tests